crates/proxy-server/src/main.rs

```diff
@@ -1,81 +1,290 @@
 use anyhow::Context;
 use clap::Parser;
+use config::CallbackSignature;
+use data_loader::KeySource;
+use kumo_server_common::diagnostic_logging::{DiagnosticFormat, LoggingConfig};
+use kumo_server_common::start::StartConfig;
 use nix::sys::resource::{getrlimit, setrlimit, Resource};
-use tokio::net::TcpListener;
+use std::path::PathBuf;
+use std::time::Duration;
 
+mod mod_proxy;
 mod proxy_handler;
 
-/// KumoProxy SOCKS5 Proxy Server
+/// KumoProxy SOCKS5 Proxy Server.
+///
+/// Full docs available at: <https://docs.kumomta.com>
 #[derive(Debug, Parser)]
-#[command(about)]
-pub struct Opt {
+#[command(about, version=version_info::kumo_version())]
+struct Opt {
+    /// Lua policy file to load.
+    /// Mutually exclusive with --listen and other legacy options.
+    #[arg(long)]
+    policy: Option<PathBuf>,
+
+    /// Directory where diagnostic log files will be placed.
+    ///
+    /// If omitted, diagnostics will be printed to stderr.
+    #[arg(long)]
+    diag_log_dir: Option<PathBuf>,
+
+    /// How diagnostic logs render. full, compact and pretty are intended
+    /// for human consumption.
+    ///
+    /// json outputs machine readable records.
+    #[arg(long, default_value = "full")]
+    diag_format: DiagnosticFormat,
+
+    // Legacy CLI options for backwards compatibility
+    // These are mutually exclusive with --policy
+    /// [Legacy] Address(es) to listen on (e.g., "0.0.0.0:5000").
+    /// Mutually exclusive with --policy.
```

Comment:
You can do something like:

```rust
#[arg(long, conflicts_with="proxy-config")]
```

to have the clap argument parser enforce the mutually exclusive/conflicting arguments.

Be sure to check that this works as you expect: I don't know if it wants `proxy-config` or `proxy_config`, and you won't know which until you try to run it--it won't catch an issue at compile time.

The same can be applied to each of the conflicting arguments.